internal/pkg/instrumentation/bpf/go.opentelemetry.io/auto/sdk/bpf/probe.bpf.c

```diff
@@ -0,0 +1,219 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "arguments.h"
+#include "go_context.h"
+#include "go_types.h"
+#include "trace/span_context.h"
+#include "trace/start_span.h"
+
+char __license[] SEC("license") = "Dual MIT/GPL";
+
+#define MAX_CONCURRENT 50
+// TODO: tune this. This is a just a guess, but we should be able to determine
+// the maximum size of a span (based on limits) and set this. Ideally, we could
+// also look into a tiered allocation strategy so we do not over allocate
+// space (i.e. small, medium, large data sizes).
+#define MAX_SIZE 1024
+
+// Injected const.
+volatile const u64 span_context_trace_id_pos;
+volatile const u64 span_context_span_id_pos;
+volatile const u64 span_context_trace_flags_pos;
+
+struct otel_span_t {
+    struct span_context sc;
+    struct span_context psc;
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__type(key, void*);
+	__type(value, struct otel_span_t);
+	__uint(max_entries, MAX_CONCURRENT);
+} active_spans_by_span_ptr SEC(".maps");
+
+struct event_t {
+	u32 size;
+	char data[MAX_SIZE];
+};
+
+struct {
+    __uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+    __uint(key_size, sizeof(u32));
+    __uint(value_size, sizeof(struct event_t));
+    __uint(max_entries, 1);
+} new_event SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+} events SEC(".maps");
+
+static __always_inline long write_span_context(void *go_sc, struct span_context *sc) {
+	if (go_sc == NULL) {
+		bpf_printk("write_span_context: NULL go_sc");
+		return -1;
+	}
+
+	long rc = 0;
+	void *tid = (void *)(go_sc + span_context_trace_id_pos);
+	long ret = bpf_probe_write_user(tid, &sc->TraceID, TRACE_ID_SIZE);
+	if (ret != 0) {
+		rc--;
+		bpf_printk("write_span_context: failed to write trace ID: %ld", ret);
```

Comment:
If we fail to write, do we want to return or keep trying for the other fields (as in the current code)?